chanbackup/recover.go

```diff
@@ -121,3 +129,122 @@ func UnpackAndRecoverMulti(packedMulti PackedMulti,
 
 	return Recover(chanBackups.StaticBackups, restorer, peerConnector)
 }
+
+// SignCloseTx produces a signed commit tx from a channel backup.
+func SignCloseTx(s Single, keyRing keychain.KeyRing, ecdher keychain.ECDHRing,
+	signer input.Signer) (*wire.MsgTx, error) {
+
+	if s.CloseTxInputs == nil {
+		return nil, errors.New("channel backup does not have data " +
+			"needed to sign force close tx")
+	}
+
+	// Each of the keys in our local channel config only have their
+	// locators populate, so we'll re-derive the raw key now.
+	var err error
+	s.LocalChanCfg.MultiSigKey, err = keyRing.DeriveKey(
```

Comment:
Should we place this into a new variable to avoid mutating the backup given to the function?

Reply:
The backup (type `Single`) is passed by value, so this line mutates a local copy of the variable.

To improve readability, I changed this line to save the key to a variable and pass it to functions `createSignDesc` and `createTaprootNonceProducer`. So it is clear that the backup variable does not change in the process.